crates/bin/cli-wallet/src/sync.rs

```diff
@@ -0,0 +1,208 @@
+use anyhow::{anyhow, Result};
+use node_client::{MeltState, MintQuoteState, NodeClient};
+use r2d2::Pool;
+use r2d2_sqlite::SqliteConnectionManager;
+use tonic::transport::Channel;
+use wallet::types::NodeUrl;
+
+const STARKNET_STR: &str = "starknet";
+
+/// Represents a node with its connection and associated data
+struct NodeSync {
+    node_client: NodeClient<Channel>,
+}
+
+impl NodeSync {
+    async fn new(pool: Pool<SqliteConnectionManager>, node_id: u32) -> Result<Self> {
+        let node_url = {
+            let db_conn = pool.get()?;
+            wallet::db::get_node_url(&db_conn, node_id)?
+                .ok_or_else(|| anyhow!("Node {} not found", node_id))?
+        };
+
+        let node_client = wallet::connect_to_node(&node_url)
+            .await
+            .map_err(|e| anyhow!("Failed to connect to node {}: {}", node_url, e))?;
+
+        Ok(Self {
+            node_client,
+        })
+    }
+}
+
+pub async fn sync_all_pending_operations(
+    pool: Pool<SqliteConnectionManager>,
+    _poll_interval: u64,
+    _timeout: u64,
+) -> Result<()> {
+    let (nodes, pending_mint_quotes, pending_melt_quotes) = {
+        let db_conn = pool.get()?;
+        let nodes = wallet::db::node::fetch_all(&db_conn)?;
```

Comment:
just read the nodes that you have business with. Meaning only the ones that have either a mint or melt quote pending